haskell/utils.bzl

```diff
@@ -0,0 +1,38 @@
+"""Misc utilities."""
+
+def produce_paths_module(name, version):
+  """Produce content of Paths_* module usually generated by Cabal. We do not
+  replicate the whole thing yet, only version definition which is the most
+  often used thing.
+
+  Args:
+    name:    String, package name.
+    version: String, version string with dot-separated components, e.g. "1.2.3".
+
+  Returns:
+    String, String:
+      * Filename for the module file.
+      * Textual contents of the file.
+  """
+
+  adjusted_pkgname = name.replace("-", "_")
+
+  contents = """
+  {-# OPTIONS_GHC -fno-warn-implicit-prelude #-}
+  module Paths_"""
```

Comment:
(drive-by comment) I'd probably go for a template with formatting here, i.e.:
```python
contents = """
{{-# OPTIONS_GHC ... #-}}
module Paths_{adjusted_pkgname} (
  version
) where

...

version = Version [{version}] []
""".format({
    "adjusted_pkgname": name.replace("-", "_"),
    "version": version.replace(".", ","),
})
```

Reply:
Well I'd probably use `ctx.action.expand_tempalte` now.